Dockerfile

```diff
@@ -23,6 +23,7 @@ ENV M2_HOME /opt/apache-maven-3.3.9
 ENV maven.home $M2_HOME
 ENV M2 $M2_HOME/bin
 ENV PATH $M2:$PATH
+RUN mkdir --parents --mode 777 /root/.mvnrepository
```

Comment:
does it need to be 777 ? just the sticky bit (+t) would be enough if it's run as another user,

Reply:
@kbsingh Can you comment? You had mentioned some reasons to make this 777 during our call.

Reply:
cc @jfchevrette. Is 777 OK?

Reply:
I think it would make more sense if maven would write the files to $HOME/.mvnrepository (/home/jenkins/.mvnrepository) instead of /root. However if we want to leave it in /root then the /root/.mvnrepository folder need gid=0 and g+rwx

Reply:
I think there is a few things refing `/root/.mvnrepository` directly i.e: 

https://github.com/search?utf8=%E2%9C%93&q=%22%2Froot%2F.mvnrepository%22&type=Code

not sure if this stuff is actually used tho,

Reply:
@chmouel good point. in that case lets leave it in /root/.mvnrepository for now

Reply:
Things that can be searched are better than the random `base64(settings.xml) > k8s.secret.yml`. There might be a lot of unknown references as well. Let's stick with this path to not break anything.

Reply:
@hrishin @jfchevrette @chmouel Good to merge?

Reply:
@jaseemabid LGTM

Reply:
@jaseemabid I think @jfchevrette was saying : 

> However if we want to leave it in /root then the /root/.mvnrepository folder need gid=0 and g+rwx

so that need to be changed before merging,